### ISSUE TITLE
A lot crashes due to Android Branch SDK 5.0.9

### ISSUE THREAD

**Issue by skinner2012** [BranchMetrics/android-branch-deep-linking-attribution#928]
I have `io.branch.sdk.android:library:5.0.9` in my Android app. 

And 8/23 I got 1.2k crashes from the same user via Firebase Crashlytics. Here is the stacktrace:

```
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'void io.branch.referral.Branch.addExtraInstrumentationData(java.lang.String, java.lang.String)' on a null object reference
       at io.branch.referral.BranchPostTask.doInBackground(BranchPostTask.java:39)
       at io.branch.referral.BranchPostTask.doInBackground(BranchPostTask.java:17)
       at android.os.AsyncTask$2.call(AsyncTask.java:292)
       at java.util.concurrent.FutureTask.run(FutureTask.java:237)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at java.lang.Thread.run(Thread.java:818)
```

<img width="1423" alt="Screen Shot 2021-08-23 at 16 53 34" src="https://user-images.githubusercontent.com/1660764/130534067-4c7d24b9-077a-44ee-938d-f320cba2a439.png">

**Comment by Jeff11** [comment]
Duplicate of https://github.com/BranchMetrics/android-branch-deep-linking-attribution/issues/913

**Comment by jf-branch** [comment]
As noted in #913

This NPE in BranchPostTask.doInBackground() should be fixed by our 5.0.12 release. Please update your SDKs to the newest version. We will be closing out this issue thread for now. Please reopen #913 if the NPE persists.
